$MirOS$
$OpenBSD: patch-Source_kwsys_SystemTools_cxx,v 1.12 2013/06/06 12:47:06 dcoppa Exp $

Don't spam systrace by attempting to create dirs which already exist.

--- Source/kwsys/SystemTools.cxx.orig	Tue Jul  2 13:41:15 2013
+++ Source/kwsys/SystemTools.cxx	Sun Aug  4 14:16:54 2013
@@ -231,11 +231,13 @@ inline void Realpath(const char *path, k
 }
 #else
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
 inline int Mkdir(const char* dir)
 {
-  return mkdir(dir, 00777);
+  struct stat sb;
+  return stat(dir, &sb) && mkdir(dir, 00777);
 }
 inline int Rmdir(const char* dir)
 {
